tests/processor/document/test_split_into_partition.py

```diff
@@ -0,0 +1,185 @@
+
+
+def test_split_document_to_partitions_with_initial_partition():
+
+    split_document_to_partitions = SplitDocumentToPartitions(
+        pattern="(<start>|<middle>|<end>)",
+        label_group_id=0,
+        collect_statistics=True,
+        initial_partition_label="<initial_part>",
+    )
+
+    document = DocumentWithPartition(text=TEXT2)
+    new_document = split_document_to_partitions(document)
+
+    partitions = new_document.partition
+    assert len(partitions) == 3
+    partition = partitions[0]
+    assert new_document.text[partition.start : partition.end] == "This is initial text."
+    assert partition.label == "<initial_part>"
+    document_lengths = split_document_to_partitions._statistics["document_lengths"]
+    partition_lengths = split_document_to_partitions._statistics["partition_lengths"]
+    assert document_lengths == [len(TEXT2)]
+    assert sum(document_lengths) == sum(partition_lengths)
+
+
+def test_split_document_to_partitions_without_label_group_id():
+    split_document_to_partitions = SplitDocumentToPartitions(
+        pattern="(<start>|<middle>|<end>)",
+        label_whitelist=["<start>", "<middle>", "<end>"],
+        skip_initial_partition=True,
+    )
+    document = DocumentWithPartition(text=TEXT1)
+    new_document = split_document_to_partitions(document)
+
+    partitions = new_document.partition
+    assert len(partitions) == 0
+
+
+def test_get_partitions_with_matcher():
+    document = DocumentWithPartition(text=TEXT1)
+    matcher = re.compile("(<start>|<middle>|<end>)").finditer
+    partitions = []
+    for partition in get_partitions_with_matcher(
+        document=document,
+        matcher=matcher,
+        label_group_id=0,
+        label_whitelist=["<start>", "<middle>", "<end>"],
+    ):
+        assert isinstance(partition, LabeledSpan)
+        for p in partitions:
+            assert not have_overlap((p.start, p.end), (partition.start, partition.end))
+        partitions.append(partition)
+
+
+def test_split_document_to_partitions_with_no_parts_added():
+    split_document_to_partitions = SplitDocumentToPartitions(
+        pattern="(<start>|<middle>|<end>)",
+        label_group_id=0,
+        label_whitelist=[],
+        skip_initial_partition=True,
+    )
+    document = DocumentWithPartition(text=TEXT1)
+    new_document = split_document_to_partitions(document)
+
+    partitions = new_document.partition
+    assert len(partitions) == 0
+
+    split_document_to_partitions = SplitDocumentToPartitions(
+        pattern="(<start>|<middle>|<end>)",
+        label_group_id=0,
+        label_whitelist=[],
+    )
+    document = DocumentWithPartition(text=TEXT1)
+    new_document = split_document_to_partitions(document)
+
+    partitions = new_document.partition
+    assert len(partitions) == 1
+    partition = partitions[0]
+    assert partition.label == "initial part"
+    assert new_document.text[partition.start : partition.end] == TEXT1
```

Comment:
Ok, this needs to be well document. The behavior looked a bit strange at the beginning, when compared to the other outputs for no matches, but I guess, it is intended. Or do you think it makes sense to change that?

Reply:
Yeah so what I thought that it would be useful if we want whole document as a partition then we can do it this way with a custom label to partition. So I think it is fine as it is at the moment. I'll add necessary documentation.

Reply:
sounds good!